src/GovUk.Education.ExploreEducationStatistics.Content.Api/Controllers/ReleaseController.cs

```diff
@@ -35,39 +43,71 @@ public async Task<ActionResult<List<ReleaseSummaryViewModel>>> ListReleases(stri
         {
-            return await _releaseCacheService.GetReleaseAndPublication(publicationSlug)
+            return await GetReleaseViewModel(publicationSlug)
                 .HandleFailuresOrOk();
         }
 
         [HttpGet("publications/{publicationSlug}/releases/latest/summary")]
         public async Task<ActionResult<ReleaseSummaryViewModel>> GetLatestReleaseSummary(string publicationSlug)
         {
-            return await _releaseCacheService.GetReleaseSummary(publicationSlug)
+            return await GetReleaseSummaryViewModel(publicationSlug)
                 .HandleFailuresOrOk();
         }
 
         [MemoryCache(typeof(GetReleaseCacheKey), durationInSeconds: 15, expiryScheduleCron: HalfHourlyExpirySchedule)]
         [HttpGet("publications/{publicationSlug}/releases/{releaseSlug}")]
         public async Task<ActionResult<ReleaseViewModel>> GetRelease(string publicationSlug, string releaseSlug)
         {
-            return await _releaseCacheService.GetReleaseAndPublication(
-                    publicationSlug,
-                    releaseSlug)
+            return await GetReleaseViewModel(publicationSlug, releaseSlug)
                 .HandleFailuresOrOk();
         }
 
         [HttpGet("publications/{publicationSlug}/releases/{releaseSlug}/summary")]
         public async Task<ActionResult<ReleaseSummaryViewModel>> GetReleaseSummary(string publicationSlug,
             string releaseSlug)
         {
-            return await _releaseCacheService.GetReleaseSummary(
+            return await GetReleaseSummaryViewModel(
                     publicationSlug,
                     releaseSlug)
                 .HandleFailuresOrOk();
         }
+
+        private Task<Either<ActionResult, ReleaseViewModel>> GetReleaseViewModel(
+            string publicationSlug,
+            string? releaseSlug = null)
+        {
+            return _publicationCacheService.GetPublication(publicationSlug)
+                .OnSuccessCombineWith(publication => _methodologyCacheService.GetSummariesByPublication(publication.Id))
+                .OnSuccessCombineWith(_ => _releaseCacheService.GetRelease(publicationSlug, releaseSlug))
+                .OnSuccess(tuple =>
+                {
+                    var (publication, methodologySummaries, release) = tuple;
+                    return new ReleaseViewModel(
+                        release,
+                        new PublicationViewModel(publication, methodologySummaries)
+                    );
+                });
+        }
+
+        private Task<Either<ActionResult, ReleaseSummaryViewModel>> GetReleaseSummaryViewModel(
+            string publicationSlug,
+            string? releaseSlug = null)
+        {
+            return _publicationCacheService.GetPublication(publicationSlug)
+                .OnSuccessCombineWith(_ => _releaseCacheService.GetRelease(publicationSlug, releaseSlug))
```

Comment:
Same as above - could perform these in parallel?

Reply:
See reply to above. 🙂